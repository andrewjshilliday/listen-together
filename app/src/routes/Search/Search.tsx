import React, { useState, useEffect } from 'react';
import { MKSearch } from '../../services/MusicKitApi';
import { MediaItemCardCarousel } from '../../components/common';
import CircularProgress from '@material-ui/core/CircularProgress';
import './Search.scss';

const Search: React.FC = (props: any) => {
  const [searchResults, setSearchResults] = useState({} as any);
  const [loading, setLoading] = useState(true);
  const searchQuery = props.match.params.query;

  useEffect(() => {
    const getSearchResults = async () => {
      setLoading(true);
      setSearchResults(await MKSearch(searchQuery, undefined, 10));
      setLoading(false);
    };
    getSearchResults();
  }, [searchQuery]);

  if (loading) return (<div className="loading"><CircularProgress /></div>);

  return (
    <div className="search view">
      <h1>Search - {searchQuery}</h1>
      {searchResults.playlists && <>
        <h2>Artists</h2>
        <MediaItemCardCarousel items={searchResults.artists.data}></MediaItemCardCarousel>
      </>}
      {searchResults.albums && <>
        <h2>Albums</h2>
        <MediaItemCardCarousel items={searchResults.albums.data}></MediaItemCardCarousel>
      </>}
      {searchResults.playlists && <>
        <h2>Playlists</h2>
        <MediaItemCardCarousel items={searchResults.playlists.data}></MediaItemCardCarousel>
      </>}
    </div>
  );
}

export default Search;
