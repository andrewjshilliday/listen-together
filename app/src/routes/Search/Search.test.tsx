import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import Search from './Search';

jest.mock('../../services/MusicKit');
jest.mock('../../services/MusicKitApi', () => ({
  MusicKitApiService: require('../../__mocks__/services/MusicKitApi/MusicKitApi.mock').default
}));

jest.useFakeTimers();

it('renders correctly', async () => {  
  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/search/gary%20clark%20jr' ]}>
      <Route path="/search/:query" component={Search} />
    </MemoryRouter>);

  jest.runTimersToTime(1);

  await act(async () => {
    await Promise.resolve();
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
