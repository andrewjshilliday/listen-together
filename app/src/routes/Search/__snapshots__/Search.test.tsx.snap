// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-fzoiQi jkvegY"
>
  <h1>
    Search - 
    gary clark jr
  </h1>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Artists
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr."
              className="img-fluid rounded"
              src="placeholder.jpeg"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Albums
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1447538299"
            onClick={[Function]}
          >
            <img
              alt="This Land"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/906127510"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr. Live"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr. Live
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1022109426"
            onClick={[Function]}
          >
            <img
              alt="The Story of Sonny Boy Slim"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Story of Sonny Boy Slim
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560639480"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1200488004"
            onClick={[Function]}
          >
            <img
              alt="Live North America 2016"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Live North America 2016
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/455141519"
            onClick={[Function]}
          >
            <img
              alt="The Bright Lights - EP"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              The Bright Lights - EP
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/560652745"
            onClick={[Function]}
          >
            <img
              alt="Blak and Blu (Deluxe Version)"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Blak and Blu (Deluxe Version)
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/626872587"
            onClick={[Function]}
          >
            <img
              alt="iTunes Session"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              iTunes Session
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr.
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1279718975"
            onClick={[Function]}
          >
            <img
              alt="Come Together - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Come Together - Single
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & Junkie XL
            </span>
          </a>
        </div>
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/album/1494805000"
            onClick={[Function]}
          >
            <img
              alt="This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              This Land (Remix) [From the Tonight Show Starring Jimmy Fallon] [feat. Black Thought] - Single
            </span>
          </a>
          <a
            href="/artist/266463936"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
            >
              Gary Clark Jr. & The Roots
            </span>
          </a>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="sc-AxirZ fJUMyY"
  >
    <h2>
      Playlists
    </h2>
    <div
      className="sc-AxiKw enaPwj"
    >
      <div
        className="sc-AxhCb iEIFwb"
        onClick={[Function]}
      />
      <div
        className="sc-AxgMl eFueIV"
      >
        <div
          className="sc-AxjAm dYpDUQ"
        >
          <a
            href="/playlist/pl.a5ef67f3dde74a568e583ac9f0406d50"
            onClick={[Function]}
          >
            <img
              alt="Gary Clark Jr. Essentials"
              className="img-fluid rounded"
            />
            <span
              className="text-truncate"
            >
              Gary Clark Jr. Essentials
            </span>
          </a>
          <span
            className="text-truncate"
          >
            Apple Music Blues
          </span>
        </div>
      </div>
      <div
        className="sc-AxhCb sc-AxhUy ixXRnl"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
